nistats/second_level_model.py

```diff
@@ -0,0 +1,381 @@
+                    if not isinstance(cond, str) and isinstance(name, str):
+                        raise ValueError('When second_level_input is a'
+                                         ' DataFrame, first_level_conditions '
+                                         'must be (str, str) pair')
+        else:
+            raise ValueError('second_level_input must be a list of'
+                             ' `FirstLevelModel` objects, a pandas DataFrame'
+                             ' or a list Niimg-like objects. Instead %s '
+                             'was provided' % type(second_level_input))
+
+        # check conditions if provided
+        if first_level_conditions is not None:
+            if isinstance(first_level_conditions, list):
+                for cidx, (name, cond) in enumerate(first_level_conditions):
+                    if not isinstance(name, str):
+                        raise ValueError('condition name at idx %d is %s '
+                                         'instead of str' % (cidx, type(name)))
+                    if not isinstance(cond, (str, np.ndarray)):
+                        raise ValueError('condition at idx %d is %s instead of'
+                                         ' str or array' %
+                                         (cidx, type(cond)))
+            else:
+                raise ValueError('first_level_conditions is not a list. '
+                                 'It is %s instead' % type(first_level_conditions))
+
+        # check regressors
+        if regressors is not None:
+            if not isinstance(regressors, pd.DataFrame):
+                raise ValueError('regressors must be a pandas DataFrame')
+            if 'model_id' not in regressors.columns:
+                raise ValueError('regressors DataFrame must contain column'
+                                 '"model_id"')
+            if len(regressors.columns) < 2:
+                raise ValueError('regressors should contain at least 2 columns'
+                                 'one called "model_id" and the other with'
+                                 'a given regressor')
+
+        # check design matrix
+        if design_matrix is not None:
+            if not isinstance(design_matrix, pd.DataFrame):
+                raise ValueError('design matrix must be a pandas DataFrame')
+
+
+        # Build the design matrix X and list of imgs Y for GLM fit
+        if isinstance(second_level_input, pd.DataFrame):
+            maps_table = second_level_input
+            # Get only first level conditions if provided
+            if first_level_conditions is not None:
+                for name, condition in first_level_conditions:
+                    if condition not in maps_table['map_name'].tolist():
+                        raise ValueError('condition %s not present in'
+                                         ' second_level_input' % condition)
+                condition_list = list(zip(*first_level_conditions))[:][1]
```

Comment:
tlooking again at this, would not be more readable:?

[condition[0] for condition in first_level_conditions]

no idea what passed my mind when I used zip here